libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackMetadata.kt

```diff
@@ -0,0 +1,190 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.location.Location
+import com.google.gson.Gson
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.utils.toTelemetryLocations
+import com.mapbox.navigation.core.telemetry.LocationsCollector
+
+/**
+ * It's the wrapper of [FeedbackMetadata] that collect locations after instance of
+ * [FeedbackMetadataWrapper] is created and stop collecting in 2 possible cases:
+ *
+ * - locations buffer is full;
+ * - [FeedbackMetadataWrapper.get] is call.
+ *
+ * [FeedbackMetadataWrapper.get] can be invoked whenever it needs.
+ */
+class FeedbackMetadataWrapper internal constructor(
```

Comment:
What I'm still missing in this structure is the ability to provide additional context to the same feedback wrapper object.

Now, if developers don't want to send feedback right away but store it (for example, on disk) after the user provided all details, they'd have to store 2 objects - first the metadata wrapper and then also an object with feedback type and details. When they withdraw the two, they need to create a mapping between them to send the right metadata with the right feedback details.

That's a lot of steps that can go wrong and possibly we could help developers by exposing an ability to store details in this same metadata object, similarly as it was done previously in the `CachedNavigationFeedbackEvent`.

Basically what I'm suggesting is adding
```
var @FeedbackEvent.Type feedbackType: String?,
var description: String?,
var @FeedbackEvent.Source feedbackSource: String?,
var screenshot: String?,
var feedbackSubType: Array<@FeedbackEvent.SubType String>? = emptyArray(),
```
to `FeedbackMetadataWrapper` and `FeedbackMetadata` or a method like `FeedbackMetadataWrapper#addFeedbackData` with these parameters so that developers can fill the data once provided from the user (wherever that is in their app flow) and store everything on disk to be sent later.

This would change:
```
fun postUserFeedback(
    @FeedbackEvent.Type feedbackType: String,
    description: String,
    @FeedbackEvent.Source feedbackSource: String,
    screenshot: String?,
    feedbackSubType: Array<@FeedbackEvent.SubType String>? = emptyArray(),
    feedbackMetadata: FeedbackMetadata? = null,
)
```
to 
```
fun postUserFeedback(
    @FeedbackEvent.Type feedbackType: String,
    description: String,
    @FeedbackEvent.Source feedbackSource: String,
    screenshot: String?,
    feedbackSubType: Array<@FeedbackEvent.SubType String>? = emptyArray(),
)
fun postUserFeedback(
    feedbackMetadata: FeedbackMetadata
)
```

Reply:
Personally, I prefer do not to do this, and that's why:
- we don't know what data should be stored and if it's should be stored at all. Why do we think that users fill all data, but not only one field, what if the flow is next: fill type, then attach subtype, then attach subtype, then attach `subtype` and there might be a lot of things that we don't know. What if for a single location customer whats to attach multiple feedbacks that for each should be stored the same metadata;
- from above, as we don't know business cases how customers want to use feedback metadata seems that provide the plainest realization is good way;
- for your example I can suggest just customer wrapper to store anything that should be:
```kotlin
class StoredCustomerFeedback (
val @FeedbackEvent.Type feedbackType: String?,
val description: String?,
val @FeedbackEvent.Source feedbackSource: String?,
val screenshot: String?,
val feedbackSubType: Array<@FeedbackEvent.SubType String>? = emptyArray(),
val feedbackMetadata: FeedbackMetadata // String might be
){
}
```
and it's easy to modify for any customer needs(for instance add data for analytic, attach multiple feedbacks for a single location, and so on)